node-wasm/src/node.rs

```diff
@@ -0,0 +1,254 @@
+        Ok(self.0.set_peer_trust(peer_id, is_trusted).await?)
+    }
+
+    /// Request the head header from the network.
+    pub async fn request_head_header(&self) -> Result<JsValue> {
+        let eh = self.0.request_head_header().await?;
+        Ok(to_value(&eh)?)
+    }
+
+    /// Request a header for the block with a given hash from the network.
+    pub async fn request_header_by_hash(&self, hash: &str) -> Result<JsValue> {
+        let hash: Hash = hash.parse()?;
+        let eh = self.0.request_header_by_hash(&hash).await?;
+        Ok(to_value(&eh)?)
+    }
+
+    /// Request a header for the block with a given height from the network.
+    pub async fn request_header_by_height(&self, height: u64) -> Result<JsValue> {
+        let eh = self.0.request_header_by_height(height).await?;
+        Ok(to_value(&eh)?)
+    }
+
+    /// Request headers in range (from, from + amount] from the network.
+    ///
+    /// The headers will be verified with the `from` header.
+    pub async fn request_verified_headers(&self, from: JsValue, amount: u64) -> Result<Array> {
+        let header =
+            from_value::<ExtendedHeader>(from).js_context("Parsing extended header failed")?;
+        let verified_headers = self.0.request_verified_headers(&header, amount).await?;
+
+        Ok(verified_headers
+            .iter()
+            .map(to_value)
+            .collect::<StdResult<_, _>>()?)
+    }
+
+    /// Get current header syncing info.
+    pub async fn syncer_info(&self) -> Result<JsValue> {
+        let syncer_info = self.0.syncer_info().await?;
+        Ok(to_value(&syncer_info)?)
+    }
+
+    /// Get the latest header announced in the network.
+    pub fn get_network_head_header(&self) -> Result<JsValue> {
+        let maybe_head_hedaer = self.0.get_network_head_header();
+        Ok(to_value(&maybe_head_hedaer)?)
+    }
+
+    /// Get the latest locally synced header.
+    pub async fn get_local_head_header(&self) -> Result<JsValue> {
+        let local_head = self.0.get_local_head_header().await?;
+        Ok(to_value(&local_head)?)
+    }
+
+    /// Get a synced header for the block with a given hash.
+    pub async fn get_header_by_hash(&self, hash: &str) -> Result<JsValue> {
+        let hash: Hash = hash.parse().js_context("parsing hash failed")?;
+        let eh = self.0.get_header_by_hash(&hash).await?;
+        Ok(to_value(&eh)?)
+    }
+
+    /// Get a synced header for the block with a given height.
+    pub async fn get_header_by_height(&self, height: u64) -> Result<JsValue> {
+        let eh = self.0.get_header_by_height(height).await?;
+        Ok(to_value(&eh)?)
+    }
+    /// Get synced headers from the given heights range.
+    ///
+    /// If start of the range is undefined (None), the first returned header will be of height 1.
+    /// If end of the range is undefined (None), the last returned header will be the last header in the
+    /// store.
```

Comment:
I was wondering for a moment if we shouldn't here mimic [`slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice), however I'm not really sure about that tho so feel free to just resolve you or @oblique.

Reply:
I'm fine with the current implementation